module/zfs/refcount.c

```diff
@@ -26,11 +26,7 @@
 #include <sys/zfs_context.h>
 #include <sys/refcount.h>
 
-#ifdef _KERNEL
-int reference_tracking_enable = FALSE; /* runs out of memory too easily */
-#else
```

Comment:
Was your intention to only enable this while debugging this change.  Are you you proposing we enable it by default?  If the memory usage is reasonable I could enabling this for ZFS_DEBUG builds, though not for production builds.